cfpb/design-system
# Scaffold out some browser tests

Sets up webdriver.io with a couple simple spec files. Consider this an experiment -- if we think this is a good idea and we write more tests we should integrate it into our `yarn test` workflow.

## Additions

- Two commands: `yarn webdriver` and `yarn webdriver-sauce`
- Two webdriver config files, one for local use, one for sauce.
- Two spec files, one that opens/closes an expandable and another that verifies the hide/show code toggle button works correctly on the docs website (because I keep breaking this feature).

## Testing

1. Pull down branch.
1. `yarn`
1. Ensure the docs website is running locally if it isn't already (`cd docs/ && yarn start`)
1. `yarn webdriver` to run the tests using chrome locally.
1. `yarn webdriver-sauce` to run the tests against four different browsers in sauce labs. Note: you'll need to define `SAUCE_USERNAME` and `SAUCE_ACCESS_KEY` env variables.

## Notes

- Running the specs through sauce shows that the hide/show code button doesn't work in IE11, I believe because babel isn't properly transpiling the `docs/assets/js/main.js` file.
